test/python/transpiler/test_unitary_synthesis.py

```diff
@@ -594,6 +622,105 @@ def test_coupling_map_unequal_durations(self, opt):
             )
         )
 
+    @combine(
+        opt_level=[0, 1, 2, 3],
+        bidirectional=[True, False],
+        dsc=(
+            "Test direction with transpile using opt_level {opt_level} on"
+            "target with multiple 2q gates with bidirectional={bidirectional}"
+        ),
+        name="opt_level_{opt_level}_bidirectional_{bidirectional}",
+    )
+    def test_coupling_unequal_duration_with_backendv2(self, opt_level, bidirectional):
+        qr = QuantumRegister(2)
+        circ = QuantumCircuit(qr)
+        circ.append(random_unitary(4, seed=1), [1, 0])
+        backend = FakeBackend5QV2(bidirectional)
+        tqc = transpile(
+            circ,
+            backend=backend,
+            optimization_level=opt_level,
+            translation_method="synthesis",
+            layout_method="trivial",
+        )
+        tqc_index = {qubit: index for index, qubit in enumerate(tqc.qubits)}
+        self.assertGreaterEqual(len(tqc.get_instructions("cx")), 1)
+        if bidirectional:
+            for _, qlist, _ in tqc.get_instructions("cx"):
+                self.assertEqual((1, 0), (tqc_index[qlist[0]], tqc_index[qlist[1]]))
+
+        else:
+            for _, qlist, _ in tqc.get_instructions("cx"):
+                self.assertEqual((0, 1), (tqc_index[qlist[0]], tqc_index[qlist[1]]))
+
+    @combine(
+        opt_level=[0, 1, 2, 3],
+        dsc=(
+            "Test direction with transpile using opt_level {opt_level} on"
+            "target with multiple 2q gates"
+        ),
+        name="opt_level_{opt_level}",
+    )
+    def test_non_overlapping_kak_gates_with_backendv2(self, opt_level):
+        qr = QuantumRegister(2)
+        circ = QuantumCircuit(qr)
+        circ.append(random_unitary(4, seed=1), [1, 0])
+        backend = FakeBackendV2()
+        tqc = transpile(
+            circ,
+            backend=backend,
+            optimization_level=opt_level,
+            translation_method="synthesis",
+            layout_method="trivial",
+        )
+        tqc_index = {qubit: index for index, qubit in enumerate(tqc.qubits)}
+        self.assertGreaterEqual(len(tqc.get_instructions("ecr")), 1)
+        for _, qlist, _ in tqc.get_instructions("ecr"):
+            self.assertEqual((1, 0), (tqc_index[qlist[0]], tqc_index[qlist[1]]))
+
+    def test_fractional_cx_with_backendv2(self):
+        """Test fractional CX gets used if present in target."""
+        qr = QuantumRegister(2)
+        circ = QuantumCircuit(qr)
+        circ.append(random_unitary(4, seed=1), [0, 1])
+        backend = FakeMumbaiV2()
+        synth_pass = UnitarySynthesis(target=backend.target)
+        tqc = synth_pass(circ)
+        tqc_index = {qubit: index for index, qubit in enumerate(tqc.qubits)}
+        self.assertGreaterEqual(len(tqc.get_instructions("rzx")), 1)
```

Comment:
I think FakeMumbaiV2 has both RZX(pi/4) and RZX(pi/6)... do both of them get used here or because we have hardcoded pi/4 in the KAK_GATE_NAMES, only pi/4 gets used?